src/workflows/airqo_etl_utils/datautils.py

```
@@ -2288,33 +2290,153 @@ def execute_data_quality_checks(
+                        device_category, data_type, data=data, frequency=frequency
+                    )
+                except Exception as e:
+                    logger.error(
+                        f"Data quality check failed for {device_category.value} {data_type.value}: {e}",
+                        exc_info=True,
+                    )
+
+            with ThreadPoolExecutor(
+                max_workers=1, thread_name_prefix="data_quality_"
+            ) as executor:
+                future = executor.submit(run_quality_checks)
```

Ruff (0.14.8)

Local variable `future` is assigned to but never used

Remove assignment to unused variable `future`

(F841)